cfme/infrastructure/resource_pool.py

```diff
@@ -59,19 +56,19 @@ def delete(self, cancel=True):
         Args:
             cancel: Whether to cancel the deletion, defaults to True
         """
-        sel.force_navigate('infrastructure_resource_pool', context=self._get_context())
+        navigate_to(self, 'Details')
         cfg_btn('Remove from the VMDB', invokes_alert=True)
         sel.handle_alert(cancel=cancel)
 
     def wait_for_delete(self):
-        sel.force_navigate('infrastructure_resource_pools')
+        navigate_to(self, 'All')
         wait_for(lambda: not self.exists, fail_condition=False,
-             message="Wait resource pool to disappear", num_sec=500, fail_func=sel.refresh)
+                 message="Wait resource pool to disappear", num_sec=500, fail_func=sel.refresh)
 
     def wait_for_appear(self):
-        sel.force_navigate('infrastructure_resource_pools')
+        navigate_to(self, 'All')
```

Comment:
This navigation is redundant, since we should navigate inside the exists, so we can remove it and optimize out a step